include/swift/SIL/SILValue.h

```diff
@@ -195,9 +285,9 @@ struct ValueOwnershipKind {
   /// that the two ownership kinds are "compatibile".
   ///
   /// The reason why we do not compare directy is to allow for
-  /// ValueOwnershipKind::None to merge into other forms of ValueOwnershipKind.
+  /// OwnershipKind::None to merge into other forms of ValueOwnershipKind.
   bool isCompatibleWith(ValueOwnershipKind other) const {
-    return bool(merge(other));
+    return *this == merge(other);
```

Comment:
If `isCompatibleWith` is supposed to mean "a merge produces a valid value", then I think you need to test that directly, like the previous implementation:

`bool(merge(other))`

The implementation here is asymmetric.

Reply:
Yea. I hit this. It was a mistake.